mm2src/docker_tests.rs

```diff
@@ -1966,49 +1977,46 @@ mod docker_tests {
             "params": {
                 "base": "MYCOIN",
                 "rel": "MYCOIN1",
-                "swap_method": "sell",
+                "swap_method": "setprice",
                 "price": 1,
-                "volume": 0.01,
+                "max": true,
             },
         })))
         .unwrap();
         assert!(!rc.0.is_success(), "trade_preimage success, but should fail: {}", rc.1);
-        let available = BigDecimal::from(0);
-        // `required = volume + fee_to_send_taker_payment + dex_fee + fee_to_send_dex_fee`,
-        // where `volume = 0.01`, `fee_to_send_taker_payment = fee_to_send_dex_fee = 0.00001`, `dex_fee = 0.0001`.
-        // Please note `dex_fee = 0.01 / 7770` < `min_dex_fee = 0.0001`, so `dex_fee = min_dex_fee = 0.0001`
-        let required = MmNumber::from("0.01") + MmNumber::from("0.00012");
-        expect_not_sufficient_balance(&rc.1, available, required.to_decimal());
-
-        // The max available value = balance (0.000015) - transaction_fee (0.00001) = 0.000005 that is less than dust (0.00001).
-        // In this case we have to receive the `NotSufficientBalance` error.
-        //
-        // vvv Fill the MYCOIN balance vvv
-
-        let low_balance = MmNumber::from("0.000015").to_decimal();
-        let (_ctx, mycoin) = utxo_coin_from_privkey("MYCOIN", priv_key.as_ref());
-        let my_address = mycoin.my_address().expect("!my_address");
-        fill_address(&mycoin, &my_address, low_balance.clone(), 30);
+        // balance(0.000015)
+        let available = MmNumber::from("0.000015").to_decimal();
+        // balance(0.000015) + transaction_fee(0.00001)
+        let required = MmNumber::from("0.00002").to_decimal();
+        expect_not_sufficient_balance(&rc.1, available, required, None);
 
+        fill_balance_functor(MmNumber::from("0.099995").to_decimal());
         let rc = block_on(mm.rpc(json!({
             "userpass": mm.userpass,
             "mmrpc": "2.0",
             "method": "trade_preimage",
             "params": {
                 "base": "MYCOIN",
                 "rel": "MYCOIN1",
-                "swap_method": "setprice",
+                "swap_method": "sell",
```

Comment:
Same question as above.

Reply:
@artemii235 I've not changed it, just moved the test that required some funds at the bottom of the test, git seems to show this change in a different way

Reply:
@Milerius I see. Did you make this change after discussion with @sergeyboyko0791 too as mentioned here? https://github.com/KomodoPlatform/atomicDEX-API/pull/1157/files#r761791640

Reply:
> Did you make this change after discussion with @sergeyboyko0791 too as mentioned here?

I modified the order of the tests after our discussion but I didn't mention this change.

What I can say is that those orders can now be hitted before the balance check in trade preimage:

- EmptyOutputs
- EmptyUtxoSet
- DeductFeeFromOutputFailed
- NotEnoughUtxos

The first cases in this unit test hit those case

Once you have enough funds, the last test checks if you have enough funds for the dex fees now.

Before with fixed fees it's was always hitting the same part of the code!

Please let me know if my change are not adapted and will change accordingly!